### ISSUE TITLE
Move schema_awaiter into a parent struct

### ISSUE THREAD

**Issue by rukai** [shotover/shotover-proxy#877]
The current design made sense when I first wrote it as it would let us avoid duplicating connections when the connection supported awaiting on schemas.

But we ended up needing to set the schema awaiter to connect directly to the cluster instead of through shotover (cant remember why exactly? lets double check why before we go ahead with this refactor)

So the original designs benefits couldnt be used.
Instead lets do something like:
```rust
struct CassandraConnection {
   ty: CassandraConnectionTy,
   schema_awaiter: Option<SessionScylla>,
}

pub enum CassandraConnection {
    #[cfg(feature = "cassandra-cpp-driver-tests")]
    Datastax (DatastaxSession),
    CdrsTokio (CdrsTokioSessionInstance),
    Scylla (SessionScylla)
}
```

**Comment by conorbros** [comment]
Makes sense, though lets call the `CassandraConnection` enum `DriverConnection`
